### PR TITLE
Add TLS support for k8s middleware

### PR DESCRIPTION
**Pull request by chrisohaver**

### CODE REVIEW

File: middleware/kubernetes/setup.go [hunk 1]
```diff
@@ -98,6 +98,27 @@ func kubernetesParse(c *caddy.Controller) (*Kubernetes, error) {
 						continue
 					}
 					return nil, c.ArgErr()
+				case "certauth":
```
Comment (miekg) [inline middleware/kubernetes/setup.go]: We should keep this in sync with `middleware/etcd/setup.go` where we've usd `tls <something>` to specify the tls parameters.

File: middleware/kubernetes/setup.go [hunk 2]
```diff
@@ -98,6 +98,27 @@ func kubernetesParse(c *caddy.Controller) (*Kubernetes, error) {
...
+				case "clientcert":
+					args := c.RemainingArgs()
+					if len(args) != 0 {
+						k8s.APIClientCert = args[0]
+						continue
+					}
+					return nil, c.ArgErr()
+				case "clientkey":
+					args := c.RemainingArgs()
+					if len(args) != 0 {
```
Comment (miekg) [inline middleware/kubernetes/setup.go]: `> 0` might read more natural

**Review by miekg — CHANGES_REQUESTED** [review]

File: middleware/kubernetes/setup.go (hunk 1 above)
Comment (chrisohaver) [reply middleware/kubernetes/setup.go]: ok - i'll look at that code and use the same order of key files.

**Comment by miekg** [comment]
LGTM. 

This may also need some updates to the README.md? (please add that to this PR)

Ideally this should also have a unit test to see if we parse this correctly (`setup_test.go`), but I don't think there is already something in `middleware/etcd` that does this, so this is something for a future PR.

**Comment by miekg** [comment]
excellent. Thanks!

**Pull request merged**